### PR TITLE
[flutter_tools] reload dart_sdk when sources update

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Ensure that the in memory dart_sdk.js is refreshed whenever the underlying content changes.

### CODE REVIEW

File: packages/flutter_tools/lib/src/build_runner/devfs_web.dart [hunk 1]
```diff
@@ -53,8 +53,9 @@ class WebAssetServer implements AssetReader {
     UrlTunneller urlTunneller,
     BuildMode buildMode,
     bool enableDwds,
-    Uri entrypoint,
-  ) async {
+    Uri entrypoint, {
+    bool testMode = false,
```
Comment (mdebbar) [inline packages/flutter_tools/lib/src/build_runner/devfs_web.dart]: Could you explain what `testMode` means?

**Review by mdebbar — APPROVED** [review]
Thanks!!

File: packages/flutter_tools/lib/src/build_runner/devfs_web.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_runner/devfs_web.dart]: Added doc comments

**Review by jonahwilliams — COMMENTED** [review]
WIP due to test failures

**Comment by jonahwilliams** [comment]
Fixed, updated to read directly from disk

**Review by mdebbar — APPROVED** [review]

File: packages/flutter_tools/lib/src/build_runner/devfs_web.dart [hunk 2]
```diff
@@ -88,7 +95,7 @@ class WebAssetServer implements AssetReader {
       final shelf.Cascade cascade = shelf.Cascade()
         .add(dwds.handler)
         .add(dwdsHandler);
-      shelf.serveRequests(httpServer, cascade.handler);
+        shelf.serveRequests(httpServer, cascade.handler);
```
Comment (mdebbar) [inline packages/flutter_tools/lib/src/build_runner/devfs_web.dart]: nit: unnecessary indent.

File: packages/flutter_tools/lib/src/build_runner/devfs_web.dart [hunk 3]
```diff
@@ -336,6 +373,9 @@ class WebDevFS implements DevFS {
   Future<DebugConnection> _cachedExtensionFuture;
   StreamSubscription<void> _connectedApps;
 
+  // The engine last modification date for local engine build.s
+  DateTime dartSdkLastModified;
```
Comment (mdebbar) [inline packages/flutter_tools/lib/src/build_runner/devfs_web.dart]: Not used anymore?

**Pull request merged**